lsp/gitlab_duo.lua

```diff
@@ -0,0 +1,440 @@
+---@brief
+---
+--- GitLab Duo Language Server Configuration for Neovim
+---
+--- https://gitlab.com/gitlab-org/editor-extensions/gitlab-lsp
+---
+--- The GitLab LSP enables any editor or IDE to integrate with GitLab Duo
+--- for AI-powered code suggestions via the Language Server Protocol.
+---
+--- Prerequisites:
+--- - Node.js and npm installed
+--- - GitLab account with Duo Pro license
+--- - Internet connection for OAuth device flow
+---
+--- Setup:
+--- 1. Run :LspGitLabDuoSignIn to start OAuth authentication
+--- 2. Follow the browser prompts to authorize
+--- 3. Enable inline completion in LspAttach event (see example below)
+---
+--- **Inline Completion Example:**
```

Comment:
I guess it wasn't clear but I was hoping that all cases would be looked at. Not just the ones I mentioned.

Reply:
@justinmk No, I was planning to do it with the other things including making gitlab_url and application_id configurable.